### ISSUE TITLE
Question: type provider to generate lenses 

### ISSUE THREAD

**Issue by chinwobble** [fsprojects/FSharpPlus#200]
### Description

If [this RFC that allows type providers to generate types from types](https://github.com/fsharp/fslang-design/blob/master/RFCs/FS-1023-type-providers-generate-types-from-types.md) ever gets implements (its been marked as started for a few years now) would it be a useful thing to create a type provider that automatically generates lenses?

**Comment by gusty** [comment]
Yes, I think it would allow us to implement an automatic lens generator, but I don't see that feature being shipped anytime soon in F#.

**Comment by gusty** [comment]
An alternative would be [Myriad](https://github.com/MoiraeSoftware/myriad) based Lens generator for this library.

Maybe @realvictorprm is interested in doing such work.
